OpenRA.Mods.Common/Traits/Power/CanPowerDown.cs

```diff
@@ -53,13 +55,19 @@ public void ResolveOrder(Actor self, Order order)
 			{
 				disabled = !disabled;
 
-				if (Info.PowerupSound != null && disabled)
-					Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Sounds", Info.PowerupSound, self.Owner.Faction.InternalName);
+			    if (Info.PowerupSound != null && disabled)
+			    {
+                    upgradeManager.RevokeUpgrade(self, "powered", this);
```

Comment:
This should not be hardcoded. Please expose it through a field in the info class (like we do with e.g. the `PowerupSound`).